client/src/main/java/org/spine3/protobuf/TypeUrl.java

```diff
@@ -0,0 +1,207 @@
+        return isTypeUrl(typeUrlOrName) ?
+               ofTypeUrl(typeUrlOrName) :
+               ofTypeName(typeUrlOrName);
+    }
+
+    private static boolean isTypeUrl(String str) {
+        return str.contains(SEPARATOR);
+    }
+
+    private static TypeUrl ofTypeUrl(String typeNameOrUrl) {
+        final String[] parts = TYPE_URL_SEPARATOR_PATTERN.split(typeNameOrUrl);
+        if (parts.length != 2 || parts[0].trim().isEmpty() || parts[1].trim().isEmpty()) {
+            throw propagate(
+                    new InvalidProtocolBufferException("Invalid Protobuf type url encountered: " + typeNameOrUrl));
+        }
+        return new TypeUrl(parts[0], parts[1]);
+    }
+
+    private static TypeUrl ofTypeName(String typeName) {
+        final TypeUrl typeUrl = KnownTypes.getTypeUrl(typeName);
+        return typeUrl;
+    }
+
+    /**
+     * Obtains the type URL of the message enclosed into the instance of {@link Any}.
+     *
+     * @param any the instance of {@code Any} containing a {@code Message} instance of interest
+     * @return a type URL
+     */
+    public static TypeUrl ofEnclosed(AnyOrBuilder any) {
+        final TypeUrl typeUrl = ofTypeUrl(any.getTypeUrl());
+        return typeUrl;
+    }
+
+    /**
+     * Obtains the type URL of the command message.
+     *
+     * <p>The passed command must have non-default message.
+     *
+     * @param command the command from which to get the URL
+     * @return the type URL of the command message
+     */
+    public static TypeUrl ofCommand(Command command) {
+        final Any any = command.getMessage();
+        checkNotDefault(any);
+        final TypeUrl typeUrl = ofEnclosed(any);
+        return typeUrl;
+    }
+
+    /**
+     * Obtains the type URL for the passed message class.
+     */
+    public static TypeUrl of(Class<? extends Message> clazz) {
+        final Message defaultInstance = com.google.protobuf.Internal.getDefaultInstance(clazz);
+        final TypeUrl result = of(defaultInstance);
+        return result;
+    }
+
+    /**
+     *
+     *
+     * @param message
+     * @return
+     */ // TODO:2016-07-08:alexander.litus: refactor
+    public static String getTypeUrlPrefix(Descriptor message) { // TODO:2016-07-08:alexander.litus: tests
+        final Descriptors.FileDescriptor file = message.getFile();
+        if (file.getPackage().equals(GOOGLE_PROTOBUF_PACKAGE)) {
+            return GOOGLE_TYPE_URL_PREFIX;
+        }
+        final String result = file.getOptions()
+                                  .getExtension(AnnotationsProto.typeUrlPrefix);
+        return result;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String value() {
+        // Expose method to other packages.
+        return super.value();
+    }
+
+    public String getPrefix() {
+        return prefix;
+    }
+
+    public String getTypeName() {
+        return typeName;
+    }
+
+    /** Returns the simple name of the Protobuf type (not fully qualified name). */
```

Comment:
It is better to give an example here. Saying “not fully qualified name” doesn't make it clearer.